Parse/src/main/java/com/parse/ParseOkHttpClient.java

```diff
@@ -146,6 +156,124 @@ public ParseOkHttpClient(int socketOperationTimeout, SSLSessionCache sslSessionC
     return okHttpRequestBuilder.build();
   }
 
+  private ParseHttpRequest getParseHttpRequest(Request okHttpRequest) {
+    ParseHttpRequest.Builder parseRequestBuilder = new ParseHttpRequest.Builder();
+    // Set method
+    switch (okHttpRequest.method()) {
+       case OKHTTP_GET:
+           parseRequestBuilder.setMethod(ParseRequest.Method.GET);
+           break;
+       case OKHTTP_DELETE:
+           parseRequestBuilder.setMethod(ParseRequest.Method.DELETE);
+           break;
+       case OKHTTP_POST:
+           parseRequestBuilder.setMethod(ParseRequest.Method.POST);
+           break;
+       case OKHTTP_PUT:
+           parseRequestBuilder.setMethod(ParseRequest.Method.PUT);
+           break;
+       default:
+           // This should never happen
+           throw new IllegalArgumentException(
+               "Invalid http method " + okHttpRequest.method());
+     }
+
+    // Set url
+    parseRequestBuilder.setUrl(okHttpRequest.urlString());
+
+    // Set Header
+    for (Map.Entry<String, List<String>> entry : okHttpRequest.headers().toMultimap().entrySet()) {
+      parseRequestBuilder.addHeader(entry.getKey(), entry.getValue().get(0));
+    }
+
+    // Set Body
+    if (okHttpRequest.body() instanceof CountingOkHttpRequestBody) {
+      CountingOkHttpRequestBody okHttpBody = (CountingOkHttpRequestBody) okHttpRequest.body();
+      parseRequestBuilder.setBody(okHttpBody.getParseHttpBody());
+    }
```

Comment:
We can guarantee the body here is always `CountingOkHttpRequestBody `, any ideas how we can   do cast here?